ground_reference_prep/4_pair_drone_with_ground.py

```
@@ -0,0 +1,183 @@
+import geopandas as gpd
+import pandas as pd
+import numpy as np
+
+
+DRONE_MISSIONS_WITH_ALT = "/ofo-share/scratch-amritha/tree-species-scratch/ofo-all-missions-metadata-with-altitude.gpkg"
+GROUND_REFERENCE_PLOTS = "/ofo-share/species-prediction-project/raw/ground-reference/ofo_ground-reference_plots.gpkg"
+OUTPUT_SAVE_PATH = "/ofo-share/scratch-amritha/tree-species-scratch/ground-reference-drone-missions-pairs.gpkg"
+
+def classify_mission(row):
+    altitude = row['mean_altitude']
+    pitch = row['camera_pitch_derived']
+    terrain_corr = row['flight_terrain_correlation_photogrammetry']
+    front_overlap = row['overlap_front_nominal']
+    side_overlap = row['overlap_side_nominal']
+
+    # Check for NaNs
+    if pd.isna(altitude) or pd.isna(pitch) or pd.isna(terrain_corr) \
+       or pd.isna(front_overlap) or pd.isna(side_overlap):
+        return 'unknown'
+
+    # Must meet terrain fidelity requirement
+    if terrain_corr <= 0.75:
+        return 'low-terrain-fidelity'
+
+    # High-Nadir requirements
+    if (110 <= altitude <= 150 and 0 <= pitch <= 10 and
+        front_overlap >= 90 and side_overlap >= 80):
+        return 'high-nadir'
+
+    # Low-Oblique requirements
+    if (60 <= altitude <= 100 and 18 <= pitch <= 38 and
+        front_overlap >= 70 and side_overlap >= 60):
+        return 'low-oblique'
+
+    return 'unclassified'
+
+
+# Helper function to get minimum value from comma-separated values
+def extract_min_overlap(val):
+    if pd.isna(val):
+        return np.nan
+    if isinstance(val, str) and ',' in val:
+        return min(float(x.strip()) for x in val.split(','))
+    try:
+        return float(val)
+    except ValueError:
```

Do you know whether this Error is thrown and if so why? For data science code that's not production, I do worry about this type of error handling hiding errors that could be worth addressing.

---

Added this earlier just to be safe, but I checked and it's not actually needed.